zerver/migrations/0470_backfill_edit_history.py

```diff
@@ -0,0 +1,28 @@
+# Generated by Django 3.2.2 on 2021-05-11 10:15
+
+from django.db import migrations
+
+
+class Migration(migrations.Migration):
+    dependencies = [
+        ("zerver", "0469_edit_history_json"),
+    ]
+
+    operations = [
+        migrations.RunSQL(
+            """
+            UPDATE zerver_message
+            SET edit_history_json = cast(edit_history as json)
+            WHERE edit_history IS NOT NULL
+            """,
+            reverse_sql="",
```

Comment:
I'm a bit worried that this is going to be a very long-running single transaction, and we may still want to do it in id-based batches.  I don't _think_ that (as written) it will try to take a write lock on every row, but it might.

For our deployment in Zulip Cloud, it would be great if this migration had a double-write window we could upgrade into.  Otherwise we have to take a short complete downtime window and do a catch-up run of:
```
UPDATE zerver_message SET edit_history_json = case(edit_history as json) WHERE edit_history IS NOT NULL and edit_history_json IS NULL
```
...before running the next migration.

Reply:
Yeah, let's do it with double-write. Rewrote the migration without dropping batched updates.